### PR TITLE
Modules typos

### PR DESCRIPTION
**Pull request by cassella**
Fix typos in modules/{dists,internal,packages,standard}

The modified error messages do not appear in any .good or .bad files (or anywhere else in the repo).

### CODE REVIEW

File: modules/internal/DefaultSparse.chpl [hunk 1]
```diff
@@ -423,7 +423,7 @@ module DefaultSparse {
     }
 
     proc dsiTargetLocales() {
-      compilerError("targetLocales is unsuppported by sparse domains");
+      compilerError("targetLocales is unsupported by sparse domains");
```
Comment (psahabu) [inline modules/internal/DefaultSparse.chpl]: Do you know if there are tests that intentionally trigger this error that would need to be updated?

File: modules/packages/Curl.chpl [hunk 2]
```diff
@@ -392,7 +392,7 @@ proc file.perform():bool {
     err = chpl_curl_perform(this._file_internal);
   }
 
-  if err then ioerror(err, "in file.peform()");
+  if err then ioerror(err, "in file.perform()");
```
Comment (psahabu) [inline modules/packages/Curl.chpl]: Same here?

File: modules/standard/DynamicIters.chpl [hunk 3]
```diff
@@ -199,7 +199,7 @@ where tag == iterKind.leader
   if c.length == 0 then halt("The range is empty");
   if nTasks == 1 then {
     if debugDynamicIters then
-      writeln("Guided Iterator: serial execution because there is not enoguh work");
+      writeln("Guided Iterator: serial execution because there is not enough work");
```
Comment (psahabu) [inline modules/standard/DynamicIters.chpl]: Again here?

**Comment by cassella** [comment]
On master (without this PR), there are no hits for these typos except the error messages themselves.  So though I did not actually run any tests, no test has any of these messages in its .good or .bad file:

```
fortytwo@magrathea:~/src/chapel (master)$ grep -r unsuppported .
./modules/internal/DefaultSparse.chpl:      compilerError("targetLocales is unsuppported by sparse domains");

fortytwo@magrathea:~/src/chapel (master)$ grep -r file.peform .
./modules/packages/Curl.chpl:  if err then ioerror(err, "in file.peform()");

fortytwo@magrathea:~/src/chapel (master)$ grep -r "not enoguh" .
./modules/standard/DynamicIters.chpl:      writeln("Guided Iterator: serial execution because there is not enoguh work");
```

**Comment by psahabu** [comment]
Okay great, thanks for checking.

**Pull request merged**